server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -165,7 +166,18 @@ public CompletionStage<Result> confirmAddress(
     ImmutableList<AddressSuggestion> suggestions =
         addressSuggestionJsonSerializer.deserialize(
             maybeAddressJson.orElseThrow(() -> new RuntimeException("Address JSON missing")));
+    return confirmAddressWithSuggestions(
+        request, applicantId, programId, blockId, inReview, selectedAddress, suggestions);
+  }
 
+  private CompletionStage<Result> confirmAddressWithSuggestions(
+      Request request,
+      long applicantId,
+      long programId,
+      String blockId,
+      boolean inReview,
+      String selectedAddress,
+      ImmutableList<AddressSuggestion> suggestions) {
```

Comment:
Note that the way github is collapsing this is confusing. `confirmAddressWithSuggestions` just contains logic I factored out of `confirmAddress`. The logic below is added to the `renderErrorOrRedirectToNextBlock` method (expand the hidden lines to see the full method).